spec/controllers/shops_controller_spec.rb

```diff
@@ -0,0 +1,16 @@
+require	'rails_helper'
+
+RSpec.describe ShopsController,	type:	:controller	do
+  describe '#show' do
+    it 'responds successfully'	do
+      shop = create(:shop)
+      get	:show, params: { id: shop.id }
+      expect(response).to have_http_status(200)
+    end
+
+    it 'responds not found when shop not found'	do
+      expect { get(:show, params: { id: 20 }) }.to raise_error('Not Found')
```

Comment:
Hardcoded ID `20` like this will cause tests fail later, "randomly", when there are more other tests. Because there is a small chance (hence randomly) that the `shop` has ID 20. 

Better like this:
```suggestion
      expect { get(:show, params: { id: Shop.maximum(:id) + 1 }) }.to raise_error('Not Found')
```